GSA-TTS/FAC
# Handle International Addresses When Consolidating General Information Validation

# Handle International Addresses When Consolidating General Information Validation

Issue: https://github.com/GSA-TTS/FAC/issues/3538

This is a PR for merging additional functionality into `2819-consolidate-general-information-schema-validation`, represented in PR https://github.com/GSA-TTS/FAC/pull/3473. 

## Changes: 
1. Schemas:
    * Always require auditor address fields
    * All auditor address fields are allowed to be blank
    * Determine form complete-ness based on whether the auditor is USA-based and the content of the relevant fields.
2. Reformat search summary to use the international address if the domestic address doesn't exist.

## How to test:
1. Switch to the main branch. Verify that using an international auditor address causes an issue at validation. 
2. Switch to this branch. Verify that the issue does no occur.
3. Attempt a few different address inputs and verify that things step forward correctly. 
    * Incomplete address info of any kind should result in the General Information section of the checklist appearing incomplete. 
    * Regular addresses should still work just fine. 
    * Saving a USA address then coming back and switching to a non-USA address should wipe the previous fields. The same applies the other way.
4. Step through to dissemination. Verify that it passes IntakeToDissemination.
5. Verify that the newly submitted audit displays the international address in search. 

See closed PR https://github.com/GSA-TTS/FAC/pull/3521 for screenshots and (maybe) additional context. 

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
